aptos-move/move-examples/dao/ft_dao/sources/ft_dao.move

```diff
@@ -0,0 +1,580 @@
+        let weights = get_token_balance(account, dao);
+        assert!(weights >= dao.min_required_proposer_voting_power, error::permission_denied(EVOTING_WEIGHTS_NOT_ENOUGH));
+
+        // verify the parameters are legit
+        let pm = property_map::new(arg_names, arg_values, arg_types);
+        assert_function_valid(function_name, &pm);
+
+        // verify the start_time is in future
+        let now = timestamp::now_seconds();
+        assert!(start_time_sec > now, error::invalid_argument(EPROPOSAL_START_TIME_SHOULD_IN_FUTURE));
+
+        let proposal = Proposal {
+            name,
+            description,
+            function_name,
+            function_args: pm,
+            start_time_sec,
+            resolution: PROPOSAL_UNRESOLVED,
+        };
+
+        let proposal_store = borrow_global_mut<Proposals>(ft_dao);
+        let proposal_id = dao.next_proposal_id + 1;
+        table::add(&mut proposal_store.proposals, proposal_id, proposal);
+        dao.next_proposal_id = proposal_id;
+    }
+
+    /// Vote with a batch of tokens. Tokens will be stored in a locker during voting period
+    public entry fun vote(
+        account: &signer,
+        ft_dao: address,
+        proposal_id: u64,
+        vote: bool,
+        amount: u64,
+    ) acquires DAO, ProposalVotingStatistics, Proposals {
+        let dao = borrow_global_mut<DAO>(ft_dao);
+        let gtoken = &dao.governance_token;
+        let proposals = borrow_global<Proposals>(ft_dao);
+
+        // assert the proposal hasn't ended
+        let proposal = table::borrow(&proposals.proposals, proposal_id);
+        let now = timestamp::now_seconds();
+        assert!(now < proposal.start_time_sec + dao.voting_duration, error::invalid_argument(EPROPOSAL_ENDED));
+
+        let prop_stats = borrow_global_mut<ProposalVotingStatistics>(ft_dao);
+        if (!table::contains(&prop_stats.proposals, proposal_id)) {
+            let vstat = VotingStatistics {
+                total_yes: 0,
+                total_no: 0,
+                votes: table::new(),
+            };
+            table::add(&mut prop_stats.proposals, proposal_id, vstat);
+        };
+        let stats = table::borrow_mut(&mut prop_stats.proposals, proposal_id);
+
+        let voter_addr = signer::address_of(account);
+        let token_id = token::create_token_id_raw(gtoken.creator, gtoken.collection, dao.governance_token.name, PROPERTY_VERSION);
+
+        // this account owns the token
+        let balance = token::balance_of(signer::address_of(account), token_id);
+        assert!(balance >= amount, error::permission_denied(ENOT_OWN_ENOUGH_VOTING_DAO_TOKEN));
+        let token = token::withdraw_token(account, token_id, amount);
+        if (vote) {
+            stats.total_yes = stats.total_yes + amount;
+        } else {
+            stats.total_no = stats.total_no + amount;
+        };
+        table::add(&mut stats.votes, voter_addr, token);
```

Comment:
Does this mean an account cannot increase their vote afterward (e.g. they acquire more tokens from somewhere)? Not a big deal but I just want to confirm this is intentional.